dap_prinz_green_jobs/pipeline/green_measures/green_measures.py

```diff
@@ -0,0 +1,233 @@
+            )
+        es.load()
+
+        es.taxonomy_skills.rename(columns={"Unnamed: 0": "id"}, inplace=True)
+
+        if (not job_advert) and (skill_list):
+            if isinstance(skill_list[0], str):
+                raw_skills = es.format_skills(skill_list)
+            raw_skills = skill_list
+        if (job_advert) and (not skill_list):
+            if type(job_advert) == dict:
+                job_advert = [job_advert]
+            raw_skills = es.get_skills([job[self.job_text_name] for job in job_advert])
+
+        job_skills, skill_hashes = es.skill_mapper.preprocess_job_skills(
+            {
+                "predictions": dict(
+                    zip(
+                        [str(uuid4()).replace("-", "") for _ in range(len(raw_skills))],
+                        [skill for skill in raw_skills],
+                    )
+                )
+            }
+        )
+
+        extracted_green_skills = sm.get_green_skill_measures(
+            es=es,
+            raw_skills=raw_skills,
+            skill_hashes=skill_hashes,
+            job_skills=job_skills,
+            skill_threshold=self.skill_threshold,
+        )
+
+        # create green measures for each job advert and save to dictionary
+        self.green_skill_measures = []
+        for i, _ in enumerate(raw_skills):
+            if "SKILL" in extracted_green_skills[i].keys():
+                green_skill_percent = (
+                    len(extracted_green_skills[i]["SKILL"])
+                    / (len(raw_skills[i]["SKILL"]) + len(raw_skills[i]["MULTISKILL"]))
+                ) * 100
+                green_skill_count = len(extracted_green_skills[i]["SKILL"])
+                green_skills = extracted_green_skills[i]["SKILL"]
+            else:
+                green_skill_percent = 0
+                green_skill_count = 0
+                green_skills = [{}]
+            self.green_skill_measures.append(
+                {
+                    "GREEN SKILL PERCENT": green_skill_percent,
+                    "GREEN SKILL COUNT": green_skill_count,
+                    "GREEN SKILLS": green_skills,
+                }
+            )
+
+        return self.green_skill_measures
+
+    def get_occupation_measures(self, job_advert: Dict[str, str]) -> List[dict]:
+        """
+        Extract measures of greenness at the occupation-level. Measures include:
+            - GREEN CATEGORY: O*NET green occupation categorisation
+            - GREEN/NOT GREEN: whether occupation name is considered green or not green
+        """
+        if type(job_advert) == dict:
+            job_advert = [job_advert]
+
+        soc_2010_4_mapper = om.create_job_title_soc_mapper(self.jobtitle_soc_data[0])
+
+        soc_green_measures_dict = (
+            self.green_soc_data[0]
+            .set_index("soc_4_2010")[["Green Category", "Green/Non-green"]]
+            .T.to_dict()
+        )
```

Comment:
as my previous comment consider the use of a `self.get_occupation_measures_called` type argument for not repeating calculations that only really need to be done once